tgg/seqr_scripts/sex_check.py

```diff
@@ -0,0 +1,367 @@
+import argparse
+import logging
+from typing import List
+import matplotlib.pyplot as plt
+
+
+import hail as hl
+
+from gnomad.utils.reference_genome import get_reference_genome
+
+
+logging.basicConfig(
+    format="%(asctime)s (%(name)s %(lineno)s): %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger("sex_check")
+logger.setLevel(logging.INFO)
+
+
+def get_chr_cov(
+    mt: hl.MatrixTable,
+    build: str,
+    chr_name: str,
+    call_rate_threshold: float = 0.25,
+    af_threshold: float = 0.01,
+    af_field: str = "AF",
+) -> hl.Table:
+    """
+    Calculate mean chromosome coverage.
+
+    :param mt: MatrixTable containing samples with chrY variants
+    :param build: Reference used, either GRCh37 or GRCh38
+    :param chr_name: Chosen chromosome. Must be either autosome (number only) or sex chromosome (X, Y)
+    :param call_rate_threshold: Minimum call rate threshold. Default is 0.25
+    :param af_threshold: Minimum allele frequency threshold. Default is 0.01
+    :param af_field: Name of field containing allele frequency information. Default is "AF"
+    :return: Table annotated with mean coverage of specified chromosome
+    """
+
+    logger.warning(
+        "This function expects chrY to be at index 23 and chrX to be at index 22 in the reference genome contigs list!"
+    )
+
+    if chr_name == "Y":
+        chr_place = 23
+    elif chr_name == "X":
+        chr_place = 22
+    else:
+        try:
+            # Chromosome index in '.contigs' list should be one less than the chromosome number
+            chr_place = int(chr_name) - 1
+        except ValueError:
+            logger.error("chr_name cannot be converted to an integer")
+            return -99
+
+    chr_name = hl.get_reference(build).contigs[chr_place]
+
+    logger.info(
+        "Filtering to chromosome (and filtering to non-par regions if chromosome is X or Y)..."
+    )
+    sex_mt = hl.filter_intervals(
+        mt,
+        [hl.parse_locus_interval(chr_name, reference_genome=build)],
+    )
+
+    if chr_place == 22:
+        sex_mt = sex_mt.filter_rows(sex_mt.locus.in_x_nonpar())
+    if chr_place == 23:
+        sex_mt = sex_mt.filter_rows(sex_mt.locus.in_y_nonpar())
+
+    # Filter to common SNVs above defined callrate (should only have one index in the array because the MT only contains biallelic variants)
+    sex_mt = sex_mt.filter_rows(sex_mt[af_field] > af_threshold)
+
+    # TODO: Make callrate filtering optional before adding code to gnomad_methods
+    sex_mt = hl.variant_qc(sex_mt)
+    sex_mt = sex_mt.filter_rows(sex_mt.variant_qc.call_rate > call_rate_threshold)
```

Comment:
```suggestion
    # Filter to common SNVs above defined callrate (should only have one index in the array because the MT only contains biallelic variants)
    # TODO: Make callrate filtering optional before adding code to gnomad_methods
    sex_mt = sex_mt.filter_rows(sex_mt[af_field] > af_threshold)
    sex_mt = hl.variant_qc(sex_mt)
    sex_mt = sex_mt.filter_rows(sex_mt.variant_qc.call_rate > call_rate_threshold)
```
(reorganizing to make the flow more intuitive)